packages/core/admin/ee/server/services/review-workflows/entity-service-decorator.js

```diff
@@ -43,9 +44,29 @@ const decorator = (service) => ({
     const data = { ...opts.data };
     if (isNull(data[ENTITY_STAGE_ATTRIBUTE])) {
       delete data[ENTITY_STAGE_ATTRIBUTE];
+      return service.update.call(this, uid, entityId, { ...opts, data });
     }
 
-    return service.update.call(this, uid, entityId, { ...opts, data });
+    const entity = await service.findOne.call(this, uid, entityId, {
+      populate: [ENTITY_STAGE_ATTRIBUTE],
+    });
+    const previousStageId = entity?.[ENTITY_STAGE_ATTRIBUTE]?.id ?? null;
+
+    const updatedEntity = await service.update.call(this, uid, entityId, { ...opts, data });
+    if (previousStageId && previousStageId !== data[ENTITY_STAGE_ATTRIBUTE]) {
+      const webhookPayload = {
```

Comment:
Hey @Marc-Roig & @nathan-pichon 

I'm struggling a bit with how to make this payload match what we have defined in CONTENT-1135. Do you know the best way to find the workflow id related to these stages?

Reply:
also here you might probably want to call `this.findOne(uid, entityId, { ... } ` instead of service.findOne.call ...